### ISSUE TITLE
Suggestion: "Automatic Quality Control"

### ISSUE THREAD

**Issue by calegendre** [nimeco/twitch-1-click-quality#1]
I would like to see a version of this that simply locks in at the lowest possible quality (forced) 160p or 360p...for when I want to chat and "know what's going on" but not use a ton of data.....the lock in doesn't always stay locked in, so I'd like to see a version of this, possibly in another release that gives the option to run at the lowest possible quality! A Twitch-bandwidth-saver of sorts..

Thanks for all your work! Be well!

**Comment by nimeco** [comment]
Having the save option enabled and selecting 160p/360p/XXXp previously, when you open a new stream without transcode, since Twitch doesn't give you the option to select 160p/360p/XXXp in this manner, it'll select whatever quality the streamer has enabled at that time, which is just one quality for non-transcode streams, then, when you go back to another streamer that has 160p/360p/XXXp, it should open with that previously selected quality from the start.

TL;DR;
Open a stream with transcode, extension save enabled, click on 160p, it should save.
Open a stream without transcode, it'll default to 720p 1080p, whatever the streamer has on.
Open a stream with transcode, it should select 160p from the start.

**Comment by Homo-Vexillues** [comment]
> Open a stream with transcode, it should select 160p from the start.

What is transcoding? Is it an option on Twitch? I also have a similar question.
Usually, I set the quality to 160p, and other streams will also open at 160p. However, some streamers have a minimum quality of 360p, and in that case, the quality switches to automatic. If I choose 360p, the others will also open at 360p.
So I really do need a feature to automatically select the minimum quality.

**Comment by nimeco** [comment]
> > Open a stream with transcode, it should select 160p from the start.
> 
> What is transcoding? Is it an option on Twitch? I also have a similar question. Usually, I set the quality to 160p, and other streams will also open at 160p. However, some streamers have a minimum quality of 360p, and in that case, the quality switches to automatic. If I choose 360p, the others will also open at 360p. So I really do need a feature to automatically select the minimum quality.

Transcoding is the conversion of video from one format to the other one, changing stuff in between them, be either quality, bitrate, resolution, codecs, etc. In Twitch's case, we refer as transcoding being able to change resolution sizes, which Twitch makes it available when you begin your stream, it depends on demand, how crowded twitch is in that moment, how big is your channel, if you changed names recently (if you do, you won't get transcode for a few days, no matter how many times you restart your stream) and, if you're a partner, transcode is always available, 100% of the time.

What happened now is that OBS implemented an option called Enhanced Broadcasting that allows you to transcode everything locally in your computer and send all the different resolutions so Twitch doesn't need to have the work load to convert them on their end, and this is frequently leaving out 160p out of the question (it all depends on the bitrate people choose, 160p is possible, but most just leave it on automatic).

I just uploaded version 1.3 which selects the nearest quality when the one saved isn't available, so, in practice, if you have 160p saved, it should default to 360p. If you have 1080p saved and someone doesn't have that resolution, it should go to 720p or whatever is the highest at that time.
